Sources/TeslaSwift/TeslaSwift.swift

```diff
@@ -923,16 +935,21 @@ extension TeslaSwift {
         }
 	}
 	
-    func request<ReturnType: Decodable, BodyType: Encodable>(_ endpoint: Endpoint, body: BodyType, completion: @escaping (Result<ReturnType, Error>) -> ()) -> Void {
+    func request<ReturnType: Decodable, BodyType: Encodable>(_ endpoint: Endpoint, body: BodyType, withExtension: String? = nil,
+                                                             completion: @escaping (Result<ReturnType, Error>) -> ()) -> Void {
 		
-		let request = prepareRequest(endpoint, body: body)
+        let request = prepareRequest(endpoint, body: body, withExtension: withExtension)
 		let debugEnabled = debuggingEnabled
-		let task = URLSession.shared.dataTask(with: request, completionHandler: {
-			(data, response, error) in
-			
-			
-			guard error == nil else { completion(Result.failure(error!)); return }
-			guard let httpResponse = response as? HTTPURLResponse else { completion(Result.failure(TeslaError.failedToParseData)); return }
+		let task = URLSession.shared.dataTask(with: request, completionHandler: { [weak self] (data, response, error) in
+            guard let self = self else { return }
+			guard error == nil else {
```

Comment:
indentation here is very broken

Reply:
Yes but I don't know why, in my Xcode it seems fine 🤔, let me check ...